internal/handler/flight.go

```diff
@@ -3,26 +3,47 @@ package handler
 import (
 	"context"
 	"fmt"
-	"strconv"
-
-	"log/slog"
-	"net/http"
-
 	"github.com/dilyara4949/flight-booking-api/internal/domain"
 	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
 	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
 	"github.com/gin-gonic/gin"
 	"github.com/google/uuid"
+	"log/slog"
+	"net/http"
+	"strconv"
 )
 
-const availableDefault = false
-
 type FlightService interface {
+	GetFlights(ctx context.Context, page, pageSize int, available bool) ([]domain.Flight, error)
 	Get(ctx context.Context, id uuid.UUID, available bool) (*domain.Flight, error)
 	Create(ctx context.Context, flight request.CreateFlight) (domain.Flight, error)
 	Delete(ctx context.Context, id uuid.UUID) error
 }
 
+const (
+	availableDefault = false
+)
+
+func GetFlights(service FlightService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		available, err := strconv.ParseBool(c.Query("available"))
```

Comment:
NIT: in future, I would suggest naming "available" to smth like "onlyAvaliable" or similar. Because semantically when available is set to false it would imply that we want only flights that are not available, instead of all. Alternatively you can have an inverse filter, meaning, that you return only available flights by default and have explicit parameter "all" to return all flights.